### ISSUE TITLE
Unexpected ESLint error "Newline required at end of file but not found."

### ISSUE THREAD

**Issue by anatoliyarkhipov** [es-analysis/plato#198]
Hi! 

I run the plato with this command:

``` bash
plato -r -d report -e .eslintrc src
```

In the `.eslintrc` I have this rule:

```
"eol-last": 2,
```

And ESLInt passes on all files in the codebase when I run it manually. But nearly every file in the plato report is marked with the "Newline required at end of file but not found." error:

![image](https://cloud.githubusercontent.com/assets/1934180/18194390/727d95c6-70f6-11e6-83d3-021b8a0617ad.png)

And here is a screenshot of the same file from IDE:

![image](https://cloud.githubusercontent.com/assets/1934180/18194420/874e1ef8-70f6-11e6-8523-b52e8dda32f6.png)

How exactly does Plato run ESLint? How is it possible that results are diffrent from the manual running for the same config?

Thanks in advance!

**Comment by JSlain** [comment]
This seems like a bug.
Here is how the source file is read before being inspected by eslint:

`var source = fs.readFileSync(file).toString().trim();`

**Comment by anatoliyarkhipov** [comment]
Thanks for the response, @JSlain! Any ideas how we can fix it?

Also, in the first message I said "nearly every file", but I checked them again, and found that ESLint was disabled for the files without the error. So actually the problem exists in every file.

Maybe it somehow relates to the fact that we use Windows for development?

**Comment by JSlain** [comment]
It's not Windows related.
Il'd replace (in lib/plato.js // line 103):

```
        var source = fs.readFileSync(file).toString().trim();
        if (!source) {
          log.info('Not parsing empty file "%s"', file);
          return;
        }
```

with:

```
        var source = fs.readFileSync(file).toString();
        if (!source.trim()) {
          log.info('Not parsing empty file "%s"', file);
          return;
        }
```

I'll make a pull request when i have time

**Comment by brentsmith-dev** [comment]
Until @JSlain 's PR is integrated (has Plato found a maintainer?) I'm working around it by using his other workaround for how Plato loads the configs
`var config = new CLIEngine().getConfigForFile(yourFile);`
and following it with
`config.rules['eol-last'] = 0;`
since we run the CLI linter anyway which should catch missing newlines.

Really hope someone with some time picks up this project. JSlain have you found a good replacement for this sort of analysis?

**Comment by JSlain** [comment]
Not really. we run our analysis tools in Jenkins, and uses checkstyle reporter to show eslint results.
I didn't fix that bug for a personal purpose, this wasn't really an issue for me.
